Swiftfin/Views/AdminDashboardView/ServerUserDeviceAccessView/ServerUserDeviceAccessView.swift

```diff
@@ -0,0 +1,177 @@
+//
+
+import Defaults
+import JellyfinAPI
+import SwiftUI
+
+struct ServerUserDeviceAccessView: View {
+
+    // MARK: - Environment
+
+    @EnvironmentObject
+    private var router: BasicNavigationViewCoordinator.Router
+
+    // MARK: - ViewModel
+
+    @ObservedObject
+    private var viewModel: ServerUserAdminViewModel
+    @ObservedObject
+    private var devicesViewModel = DevicesViewModel()
+
+    // MARK: - State Variables
+
+    @State
+    private var tempPolicy: UserPolicy
+    @State
+    private var error: Error?
+    @State
+    private var isPresentingError: Bool = false
+
+    // MARK: - Current Date
+
+    @CurrentDate
+    private var currentDate: Date
+
+    // MARK: - Initializer
+
+    init(viewModel: ServerUserAdminViewModel) {
+        self.viewModel = viewModel
+        self.tempPolicy = viewModel.user.policy ?? UserPolicy()
+    }
+
+    // MARK: - Body
+
+    var body: some View {
+        contentView
+            .navigationTitle(L10n.deviceAccess)
+            .navigationBarTitleDisplayMode(.inline)
+            .navigationBarCloseButton {
+                router.dismissCoordinator()
+            }
+            .topBarTrailing {
+                if viewModel.backgroundStates.contains(.updating) {
+                    ProgressView()
+                }
+                Button(L10n.save) {
+                    if tempPolicy != viewModel.user.policy {
+                        viewModel.send(.updatePolicy(tempPolicy))
+                    }
+                }
+                .buttonStyle(.toolbarPill)
+                .disabled(viewModel.user.policy == tempPolicy)
+            }
+            .onReceive(viewModel.events) { event in
+                switch event {
+                case let .error(eventError):
+                    UIDevice.feedback(.error)
+                    error = eventError
+                    isPresentingError = true
+                case .updated:
+                    UIDevice.feedback(.success)
+                    router.dismissCoordinator()
+                }
+            }
+            .alert(
+                L10n.error.text,
+                isPresented: $isPresentingError,
+                presenting: error
+            ) { _ in
+                Button(L10n.dismiss, role: .cancel) {}
+            } message: { error in
+                Text(error.localizedDescription)
+            }
+            .onFirstAppear {
+                devicesViewModel.send(.getDevices)
+            }
+    }
+
+    // MARK: - Content View
+
+    @ViewBuilder
+    var contentView: some View {
+        List {
+            Section(L10n.access) {
+                Toggle(
+                    L10n.enableAllDevices,
+                    isOn: $tempPolicy.enableAllDevices.coalesce(false)
+                )
+            }
+
+            if tempPolicy.enableAllDevices == false {
+                Section {
+                    ForEach(devicesViewModel.devices, id: \.self) { device in
+                        Toggle(
```

Comment:
For these rows, I would rather reuse the `DevicesView.DevicesRow` and refactor it a bit so that the `onDelete` action is optional and if not defined, then it isn't able to be deleted.